helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -568,6 +568,71 @@ public void createEphemeral(final String path, final Object data)
     create(path, data, CreateMode.EPHEMERAL);
   }
 
+  /**
+   * Creates a session aware ephemeral node.
```

Comment:
Please explain what "session aware" means here as well?

Reply:
Yes, will do.

Reply:
"Creates an ephemeral node that is session aware.
If the ephemeral node is created successfully, returns true.
If session is expired, returns false.
If connection is timed out or interrupted, exception is thrown."

The description is not very accurate. The gist is that given a session id, we need to make sure the created ephemeral node must be in this session. Otherwise, the creation should fail. 
ZkClient has a auto-reconnect feature. This feature makes this create ephemeral node in this session operation not that easy. 
We need handle session expired, disconnected state change. In additional close(). 

So all in all the following semantic should be true. 

"
given a session ID, the api should return true it can create an ephemeral node in this session.
If met irrecoverable condition, the API should return false.
If met recoverable  condition, the API should retry and create the ephemeral node.
"

Here, irrecoverable condition can be: session expired, new session established, zkclient closed, interrrupted, keeper exceptions such as invalid parameters etc.

Recoverable condition can be: session disconnect. (Note, session can still be reconnected with the same session id potentially later.)

Reply:
Right, thanks for the meaningful description. This is what I am planning to complete. Once the code itself is agreed by us, I will complete the description.